src/configs/configAxios.ts

```diff
@@ -0,0 +1,12 @@
+import axios, { AxiosInstance, AxiosRequestConfig } from "axios"
+import { environment } from "~/src/configs/environment"
+
+const createApiClient = (config: AxiosRequestConfig = {}): AxiosInstance => {
+  const http = axios.create(config)
+  http.interceptors.request.use(async (config) => await config)
```

Comment:
*await config* is not necessary (config is not a promise anyway. Since you are not changing any config, this interceptor is not necessary anyway.